src/frontend/src/lib/components/signer/SignerPermissions.svelte

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+	import { IconWallet } from '@dfinity/gix-components';
+	import {
+		ICRC25_PERMISSION_GRANTED,
+		ICRC27_ACCOUNTS,
+		type IcrcScope,
+		type IcrcScopedMethod,
+		type PermissionsConfirmation
+	} from '@dfinity/oisy-wallet-signer';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { type ComponentType, getContext } from 'svelte';
+	import { fade } from 'svelte/transition';
+	import { icrcAccountIdentifierText } from '$icp/derived/ic.derived';
+	import IconAstronautHelmet from '$lib/components/icons/IconAstronautHelmet.svelte';
+	import IconShield from '$lib/components/icons/IconShield.svelte';
+	import SignerOrigin from '$lib/components/signer/SignerOrigin.svelte';
+	import ButtonGroup from '$lib/components/ui/ButtonGroup.svelte';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { SIGNER_CONTEXT_KEY, type SignerContext } from '$lib/stores/signer.store';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { shortenWithMiddleEllipsis } from '$lib/utils/format.utils';
+	import { replaceOisyPlaceholders } from '$lib/utils/i18n.utils';
+
+	const {
+		permissionsPrompt: { payload, reset: resetPrompt }
+	} = getContext<SignerContext>(SIGNER_CONTEXT_KEY);
+
+	let scopes: IcrcScope[] | undefined;
+	$: scopes = $payload?.requestedScopes;
```

Comment:
I saw that in the script you are "coalescing" `scopes` when you use it (you never analyze the case where scopes is `nullish`, I mean). Would it make sense to do it directly here?

```
let scopes: IcrcScope[];
$: scopes = $payload?.requestedScopes ?? []
```

Reply:
Sure, done!